addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1095,6 +1101,66 @@ var SnippetsMenu = Widget.extend({
+     * css animation (with animation previews from 'website_animate' among others).
+     *
+     * @private
+     * @param {jQuery} $snippet
+     * @param {number} [animationsCountDelta=0] The delta to apply on the animations count as integer
+     */
+    manageOverlayVisibilityTowardsSnippetAnimations: function ($snippet, animationsCountDelta = 0) {
+        const positionalId = this._getPositionalId($snippet);
+        if (!(positionalId in this.overlayRunningAnimationCounters)) {
+            this.overlayRunningAnimationCounters[positionalId] = 0;
+        }
+        this.overlayRunningAnimationCounters[positionalId] += animationsCountDelta;
+        this.overlayRunningAnimationCounters[positionalId] = Math.max(
+            0,
+            this.overlayRunningAnimationCounters[positionalId]
+        );
+        if (this.overlayRunningAnimationCounters[positionalId] > 0) {
+            if (animationsCountDelta > 0) {
+                const maxDuration = this._getElementMaxAnimationAndTransitionDuration($snippet);
+                const timeout = Date.now() + maxDuration;
+                this.$snippetEditorArea.addClass('invisible');
+                // The normal case would be to have the transitionend event to be
+                // fired but we cannot rely on it, so we use a timeout as fallback.
+                if (timeout > this.overlayRunningAnimationNextTimeout) {
+                    clearTimeout(this.overlayPostAnimationHandler);
+                    clearTimeout(this.overlayRunningAnimationTimeoutHandler);
+                    this.overlayRunningAnimationTimeoutHandler = setTimeout(
+                        () => {
+                            this.manageOverlayVisibilityTowardsSnippetAnimations(
+                                $snippet,
+                                -this.overlayRunningAnimationCounters[positionalId]
+                            );
+                        },
+                        maxDuration
+                    );
+                    this.overlayRunningAnimationNextTimeout = timeout;
+                }
+            }
+        } else {
+            delete this.overlayRunningAnimationCounters[positionalId];
+        }
+        if (_.isEmpty(this.overlayRunningAnimationCounters)) {
+            this.overlayRunningAnimationNextTimeout = 0;
+            // Using a Timeout here prevents the cancel animation events
+            // from occurring when changing rapidly the animation (which triggered overlay flicker).
+            clearTimeout(this.overlayRunningAnimationTimeoutHandler);
+            clearTimeout(this.overlayPostAnimationHandler);
+            this.overlayPostAnimationHandler = setTimeout(
+                () => {
+                    // Force calling cover() on all overlays by triggering resize on the window.
+                    $(window).trigger("resize");
+                    this.$snippetEditorArea.removeClass('invisible');
```

Comment:
Name the class following our conventions

Reply:
This is the bootstrap class

Reply:
Oh, ok, did not know about it :+1: